web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,376 @@
+    },
+
+    toGraphData(data) {
+      const style = getComputedStyle(document.body);
+      const colorName = this.type === 'commits' ? '--color-primary-alpha-60' : (this.type === 'additions' ? '--color-green-badge-hover-bg' : '--color-red-badge-hover-bg');
+      const color = style.getPropertyValue(colorName).trim();
+      return {
+        datasets: [
+          {
+            data: data.map((i) => {
+              return {x: i.week, y: i[this.type]};
+            }),
+            pointRadius: 0,
+            pointHitRadius: 0,
+            fill: 'start',
+            backgroundColor: color,
+            borderWidth: 0,
+            tension: 0.3,
+          },
+        ],
+      };
+    },
+
+    updateOtherCharts(event, reset) {
+      const minVal = event.chart.options.scales.x.min;
+      const maxVal = event.chart.options.scales.x.max;
+      if (reset) {
+        this.dateFrom = this.startDate;
+        this.dateUntil = this.endDate;
+        this.sortContributors();
+      } else if (minVal) {
+        this.dateFrom = new Date(minVal);
+        this.dateUntil = new Date(maxVal);
+        this.sortContributors();
+      }
+    },
+
+    getOptions(type) {
+      return {
+        responsive: true,
+        maintainAspectRatio: false,
+        animation: false,
+        onClick: (e) => {
+          if (type === 'main') {
+            e.chart.resetZoom();
+            this.updateOtherCharts(e, true);
+          }
+        },
+        plugins: {
+          legend: {
+            display: false,
+          },
+          zoom: {
+            pan: {
+              enabled: true,
+              mode: 'x',
+              threshold: 20,
+              onPanComplete: this.updateOtherCharts,
+            },
+            limits: {
+              x: {
+                /* Check https://www.chartjs.org/chartjs-plugin-zoom/latest/guide/options.html#scale-limits
+                   to know what each option means */
+                min: 'original',
+                max: 'original',
+
+                // number of milliseconds in 2 weeks. Minimum x range will be 2 weeks when you zoom on the graph
+                minRange: 2 * 7 * 24 * 60 * 60 * 1000,
+              },
+            },
+            zoom: {
+              wheel: {
+                enabled: type === 'main',
+              },
+              pinch: {
+                enabled: type === 'main',
+              },
+              mode: 'x',
+              onZoomComplete: this.updateOtherCharts,
+            },
+          },
+        },
+        scales: {
+          x: {
+            type: 'time',
+            grid: {
+              display: false,
+            },
+            time: {
+              minUnit: 'day',
+            },
+          },
+          y: {
+            min: 0,
+            max: type === 'main' ? this.maxMainGraph() : this.maxContributorGraph(),
+          },
+        },
+      };
+    },
+  },
+};
+</script>
+<style scoped>
+.main-graph {
+  height: 380px;
+}
+</style>
```

Comment:
Re-order the tags in the file to script -> template -> style please. We recently changed our ordering in vue files.

Reply:
[Done](https://github.com/go-gitea/gitea/pull/25439/commits/e83b5625000149ae7ed3e2e4436b0f8b54bb2c45)